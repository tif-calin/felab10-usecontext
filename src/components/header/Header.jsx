import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => {

  return (
    <header className="Header">
      <Link to="/"><h1>Lists of Things</h1></Link>
      
      <button>☀️/🌑</button>
    </header>
  );
};

export default Header;
