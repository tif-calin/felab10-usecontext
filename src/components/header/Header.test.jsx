import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { MemoryRouter } from 'react-router-dom';

describe('Header component', () => {

  afterEach(() => cleanup());

  test('renders Header', async () => {
    const { asFragment } = await render(<MemoryRouter>
      <Header/>
    </MemoryRouter>);

    expect(asFragment()).toMatchSnapshot();
  });
  
});
