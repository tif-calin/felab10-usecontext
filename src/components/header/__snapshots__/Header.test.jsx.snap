// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <h1>
      Lists of Things
    </h1>
    <nav>
      <a
        href="/"
      >
        home
      </a>
    </nav>
  </header>
</DocumentFragment>
`;
